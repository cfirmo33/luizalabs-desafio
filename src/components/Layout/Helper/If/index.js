import PropTypes from 'prop-types'

const If = ({ test, children }) => {
  if (test) {
    return children
  }
  return false
}


If.propTypes = {
  test: PropTypes.bool.isRequired,
  children: PropTypes.any.isRequired,
}

If.defaultTypes = {
  test: false,
}
export default If
