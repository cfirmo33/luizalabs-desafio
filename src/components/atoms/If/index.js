import React from 'react'
import PropTypes from 'prop-types'

const If = ({ test, children }) => {
  if (test) {
    return children
  }
  return false
}


If.propTypes = {
  test: PropTypes.bool.isRequired,
  children: PropTypes.element.isRequired,
}

export default If
