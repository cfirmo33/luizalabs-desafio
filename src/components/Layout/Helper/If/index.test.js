import React from 'react'
import { shallow } from 'enzyme'
import If from './index'

describe('If', () => {
  it('Deve renderizar corretamente', () => {
    let wrapper = shallow(
      <If test>
        <h1>Verdadeiro</h1>
      </If>
    )
    expect(wrapper).toMatchSnapshot()

    wrapper = shallow(
      <If test={false}>
        <h1>Falso</h1>
      </If>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
