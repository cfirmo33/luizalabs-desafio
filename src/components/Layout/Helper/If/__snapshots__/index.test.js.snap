// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If Deve renderizar corretamente 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <If
    test={true}
>
    <h1>
        Verdadeiro
    </h1>
</If>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Verdadeiro",
    },
    "ref": null,
    "rendered": "Verdadeiro",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Verdadeiro",
      },
      "ref": null,
      "rendered": "Verdadeiro",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`If Deve renderizar corretamente 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <If
    test={false}
>
    <h1>
        Falso
    </h1>
</If>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
